parser_library/src/analyzer.cpp

```diff
@@ -38,6 +35,13 @@ analyzing_context& analyzer_options::get_context()
         auto h_ctx = std::make_shared<context::hlasm_context>(file_loc,
             std::move(std::get<asm_option>(ctx_source)),
             ids_init ? std::move(ids_init) : std::make_shared<context::id_storage>());
+
+        for (auto&& [name, func] : external_functions)
+        {
+            if (!h_ctx->add_external_function(name, std::move(func)))
+                throw std::logic_error("Duplicate external function: " + name);
```

Comment:
I might've missed the test but do you want to add one for the duplicate functions?